docs/Usage/RemoteKernelConnection.md

```diff
@@ -46,75 +46,58 @@ jupyter notebook --port=8888
-
-## Example Jupyter Docker Stack kernel gateway
-
-Follow this if you only need to have a simple environment to run commands inside a Docker container and nothing more.
-
-If you need to customize a Docker image (e.g. for web development) follow the section below: [**Example Docker kernel gateway**](#example-docker-kernel-gateway).
 
 ### Dockerfile
 
-- Create a `Dockerfile` based on one of the [Jupyter Docker Stacks](https://github.com/jupyter/docker-stacks).
-- Install `jupyter_kernel_gateway` in your `Dockerfile`
+- Create a `Dockerfile` based on either one of the [Jupyter Docker Stacks](https://github.com/jupyter/docker-stacks) (recommended), or your own
+- Ensure `jupyter` and `tini` are installed, either in the image or as an additional command in the `Dockerfile`
 - Expose the gateway port, in this example it will be `8888`
-- Make the command to run be the Kernel Gateway:
+- Set the `CMD` instruction to starting a notebook server, run through an init manager such as `tini`:
 
 ```Dockerfile
+# If using your own Docker image, use the following `FROM` command syntax substituting your image name
 FROM jupyter/minimal-notebook
 
-RUN pip install jupyter_kernel_gateway
-
-EXPOSE 8888
-CMD ["jupyter", "kernelgateway", "--KernelGatewayApp.ip=0.0.0.0", "--KernelGatewayApp.port=8888"]
-```
+ADD https://github.com/krallin/tini/releases/download/v0.14.0/tini /tini
+RUN chmod +x /tini
 
-### Run Docker Container with Docker commands
+# If using your own Docker image without jupyter installed:
+# RUN pip install jupyter
 
-**Note**: alternatively, see below for `docker-compose` instructions.
-
-- Build your container:
-
-```bash
-docker build -t hydro-kernel-gateway .
-```
+ENV JUPYTER_TOKEN=my_secret_token  # you can also pass this at runtime
 
-- Run your container mapping the port of the gateway
-- Give your container a name
-
-```bash
-docker run -it --rm --name hydro-kernel-gateway -p 8888:8888 hydro-kernel-gateway
+EXPOSE 8888
+ENTRYPOINT [/tini, --]
+CMD [jupyter-notebook]
+# if running as root, you need to explicitly allow this:
+# CMD [jupyter-notebook, --alow-root]
 ```
 
-**Note**: you will only be able to run one container using that port mapping. So, if you had another container using that port, you will have to stop that one first. Or alternatively, you can create a mapping to a new port and add that configuration in the Hydrogen settings (see below).
-
-### Run Docker Container with Docker Compose
+### Run with Docker Compose
 
 - Create a `docker-compose.yml` file with something like:
 
 ```yml
 version: '2'
 services:
-  hydro-kernel-gateway:
+  hydro:
     build: .
+    entrypoint: [/tini, --]
+    command: [jupyter-notebook, --alow-root]
     ports:
-      - "8888:8888"
+      - 8888:8888
+    environment:
+      # if you can also have this var in your environment it will override this
```

Comment:
This is a bit unclear because you have 2x "this" and 1x "it", and I'm not sure what they refer to.